src/Admin/AdherentAdmin.php

```diff
@@ -415,11 +424,26 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
         ;
     }
 
+    /**
+     * @param Adherent $object
+     */
+    public function preUpdate($object)
+    {
+        $em = $this->getModelManager()->getEntityManager($this->getClass());
+        $original = $em->getUnitOfWork()->getOriginalEntityData($object);
+        $this->emailSubscriptionChanged = count(array_diff($original['emailsSubscriptions'], $object->getEmailsSubscriptions())) > 0
```

Comment:
1) what about use `setSubject` method for storing the oldSubject (before update)?
check CRUDController of Sonata Admin, lines 308 and 322, the controller calls `admin::setSubject` twice with old and new object, you can create a new property of admin class `oldSubject` and
```

public function setSubject(subject) {
  if (null === $this->oldSubject) {
    $this->oldSubject = clone $subject;
  }
  parent::setSubject($subject);
}
```

2) you can use the oldSubject property in `postUpdate` (not preUpdate) method
3) you check twice the diff between before and after states of subscriptions: first in this class, and second in `AdherentEmailSubscriptionListener`. I think we can do this better with one listener `AdherentUpdateListener`. It will listen `ADHERENT_UPDATE` event and will make the check of before and after states of adherent subscriptions. you need to dispatch this event.
4) I think to know if subscription is changed, you need to check:
 - count(beforeEmailSubscriptions)
 - count(afterEmailSubscriptions)
 - count(array_intersect(beforeEmailSubscriptions, afterEmailSubscriptions))
if the three count values is different, so it's changed